dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/kcp/ComposeNavHostCompilationTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.datadog.gradle.plugin.kcp
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.intellij.lang.annotations.Language
+import org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@OptIn(ExperimentalCompilerApi::class)
+class ComposeNavHostCompilationTest : KotlinCompilerTest() {
+
+    @Test
+    fun `M inject 'NavigationViewTrackingEffect' W found nav host`() {
+        // Given
+        val navHostTestCaseSource = SourceFile.kotlin(
+            NAV_HOST_TEST_CASE_FILE_NAME,
+            NAV_HOST_TEST_CASE_SOURCE_FILE_CONTENT
+        )
+
+        // When
+        val result = compileFile(
+            target = navHostTestCaseSource,
+            deps = dependencyFiles
+        )
+        executeClassFile(
+            result.classLoader,
+            "com.datadog.android.instrumented.NavHostTestCase",
+            "NavHostTestCase",
+            listOf()
```

Comment:
nit: should it better be `emptyListOf()`?

```suggestion
            emptyListOf()
```